enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -67,28 +74,60 @@ export const useInitData = ({
   useEffect(() => {
     api.basename = config.metabaseInstanceUrl;
 
-    if (config.authType === "jwt") {
+    if (config.authType === "jwt" && config.jwtProviderUri) {
       api.onBeforeRequest = () =>
         dispatch(getOrRefreshSession(config.jwtProviderUri));
       api.sessionToken = sessionTokenState?.token?.id;
+      dispatch(setLoginStatus({ status: "initialized" }));
```

Comment:
note: make sure to include logic from #41694 if there is a merge conflict.

Reply:
Another note is that in the current version, I noticed the redux actions for SDK tokens are dispatched multiple times when I tested the SDK in my demo / reference app. I wonder if this is intentional (i.e. for refreshing the access tokens), or a re-rendering issue? Would this also be improved by this PR?

<img src="https://github.com/metabase/metabase/assets/4714175/e9a6f592-c9d4-45a7-a30a-2dce4ac2a8e7" width="200" />

Reply:
I see! This is not a priority yes, so let's worry about this later :)